tensorflow/python/ops/array_ops.py

```diff
@@ -1917,7 +1917,6 @@ def one_hot(indices, depth, on_value=1, off_value=0,
             axis, dtype], name, "one_hot") as name:
     on_value = ops.convert_to_tensor(on_value, dtype=dtype, name="on_value")
```

Comment:
Ok, I'll comment out complex dtype tests for now, and look to fix cast in another PR.